app/com/criteo/drelephant/heuristics/GarmadonTransferHeuristic.java

```diff
@@ -0,0 +1,114 @@
+
+    private static final String SELECT_ALL_DISTINCT_APP_SQL = "SELECT DISTINCT yarn_app_result_id FROM " + HEURISTIC_RESULT_TABLENAME
+            + " WHERE ready = 1";
+    private static final String SELECT_HEURISTIC_RESULT_SQL = "SELECT * FROM " + HEURISTIC_RESULT_TABLENAME
+            + " WHERE yarn_app_result_id = :yarn_app_result_id";
+    private static final String DELETE_HEURISTIC_RESULT_SQL = "DELETE FROM " + HEURISTIC_RESULT_TABLENAME
+            + " WHERE yarn_app_result_id = :yarn_app_result_id";
+    private static final String SELECT_HEURISTIC_RESULT_DETAILS_SQL = "SELECT * FROM " + HEURISTIC_RESULT_DETAILS_TABLENAME
+            + " WHERE yarn_app_heuristic_result_id = :yarn_app_heuristic_result_id";
+    private static final String DELETE_HEURISTIC_RESULT_DETAILS_SQL = "DELETE FROM " + HEURISTIC_RESULT_DETAILS_TABLENAME
+            + " WHERE yarn_app_heuristic_result_id = :yarn_app_heuristic_result_id";
+
+    public static void transfer() {
+        // Select all app in garmadon table
+        LOGGER.debug("Select all distinct ready yarn_app_result_id from " + HEURISTIC_RESULT_TABLENAME);
+        List<SqlRow> rows = Ebean.createSqlQuery(SELECT_ALL_DISTINCT_APP_SQL).findList();
+
+        for (SqlRow row : rows) {
+            String yarn_app_result_id = row.getString("yarn_app_result_id");
+            AppResult appResult = Ebean.find(AppResult.class, yarn_app_result_id);
+            if (appResult != null) {
+                LOGGER.debug("Insert data for " + yarn_app_result_id);
+                insertData(appResult, yarn_app_result_id);
+            }
+        }
+    }
+
+    private static void insertData(AppResult appResult, String yarn_app_result_id) {
+        // Set severity
+        Severity worstSeverity = appResult.severity;
+
+        Transaction transaction = null;
+        try {
+            transaction = Ebean.beginTransaction();
+
+            List<SqlRow> rows = Ebean.createSqlQuery(SELECT_HEURISTIC_RESULT_SQL)
+                    .setParameter("yarn_app_result_id", yarn_app_result_id)
+                    .findList();
+            for (SqlRow row : rows) {
+                AppHeuristicResult appHeuristicResult = new AppHeuristicResult();
+                String id = row.getString("id");
+                appHeuristicResult.heuristicClass = Utils.truncateField(row.getString("heuristic_class"),
+                        AppHeuristicResult.HEURISTIC_CLASS_LIMIT, id);
+                appHeuristicResult.heuristicName = Utils.truncateField(row.getString("heuristic_name"),
+                        AppHeuristicResult.HEURISTIC_NAME_LIMIT, id);
+                appHeuristicResult.score = row.getInteger("score");
+                appHeuristicResult.severity = Severity.byValue(row.getInteger("severity"));
+
+                List<SqlRow> detail_rows = Ebean.createSqlQuery(SELECT_HEURISTIC_RESULT_DETAILS_SQL)
```

Comment:
can you make dedicated methods for AppHeuristicResult extraction and AppHeuristicResultDetails extraction ?